### DISCUSSION TITLE
Advanced setup questions

### DISCUSSION THREAD

**Discussion by f8vze** [RainbowMiner/RainbowMiner#1680] [Q&A] [answered]
I am a new user. Have been learning (reading documentation) and setting up 4 rigs. 

I can promote RainbowMiner to others if I get it to work well. I have some questions;

1) The estimate income per day does not match the pool balances after a few days? For example, 10 USD estimated x 4 rigs should give 40 approx USD/day whereas the actual is much less (20-25% thereof estimate). Also see question 2 in this regard.
2) The pool balances on the 4 rigs (all setup as standalone with copied ./Config files and ./setup.sh executed) do not match, even though the config was just copied in full. One machine shows twice as much as another etc. Full config is the same on all machines. Could it be that pools are blocking this sort of setup?
3) I have now tried to change to 1 master/3 slaves setup to see if it will improve the situation in question 2.
4) Checking the 'best pools' I noticed that it contains wallet ID's that are not mine, like BTC 3PfUUT1Tknfyd4SnYrEwwpUEAQEzWd2BuD, and apparently these are coded into the config (https://github.com/RainbowMiner/RainbowMiner/blob/master/Data/dconf.json)? Could this be related to missing income in question 1? Is this for the benchmarking?
5) Why do quite a few algorithms show as, for example, 10+ BTC per GH per day in the Best Pools overview (and much more still)? I assume these are unrealistic numbers?
6) How does the transfer of earned pool work funds to the wallet of choice (XMR for me) work? Are somehow all these coin mining automatically converted? I can see some pool balances (30 USD for example atm) - when and how will these be transferred into XMR?
7) Is there a way to make the web interface more dense/compact (config file edit is fine!) and is it possible to see all client servers to a master server in one view without using rbminer?
8) To enable all pools and all mining algo's does one simply need to have a BTC wallet and then enable all pools, or are multiple wallets required? Does adding more coins make the overall algorithm more profitable somehow (more options available)?
9) Is the "fast switching" option I saw in the config files of interest to use?

Thank you

**Comment by RainbowMiner** [comment, marked as answer]
> 1) The estimate income per day does not match the pool balances after a few days? For example, 10 USD estimated x 4 rigs should give 40 approx USD/day whereas the actual is much less (20-25% thereof estimate). Also see question 2 in this regard.
> 2) The pool balances on the 4 rigs (all setup as standalone with copied ./Config files and ./setup.sh executed) do not match, even though the config was just copied in full. One machine shows twice as much as another etc. Full config is the same on all machines. Could it be that pools are blocking this sort of setup?
> 3) I have now tried to change to 1 master/3 slaves setup to see if it will improve the situation in question 2.

It should. In Standalone mode every miner is solely responsible for it's own database. It may happen, that they seem async with balances, since the pool APIs will be called every hour, only (the value of the balance-interval can be changed via config.txt). The estimated income heavily depends on time and value of the respective coins on the different exchanges. It's always a moment-snapshot, so there may be variations. Regard it as a rough estimate.

> 4) Checking the 'best pools' I noticed that it contains wallet ID's that are not mine, like BTC 3PfUUT1Tknfyd4SnYrEwwpUEAQEzWd2BuD, and apparently these are coded into the config (https://github.com/RainbowMiner/RainbowMiner/blob/master/Data/dconf.json)? Could this be related to missing income in question 1? Is this for the benchmarking?

Pretty right - there are two situations, when the best pools have my wallet addresses.
- once a day, the rigs connect to these pools for some minutes to support my work. The value can be changed in config.txt
- if you have MiningRigRentals enabled, RainbowMiner per default will try to benchmark all supported algorithms. With that, you do not need to setup your own pools for all the different kind of algorithms, but can benefit from rentals thereof.

> 5) Why do quite a few algorithms show as, for example, 10+ BTC per GH per day in the Best Pools overview (and much more still)? I assume these are unrealistic numbers?

No, these numbers are for real. The secret is to look at the "per GH". Let's look at "BeamHash3" on Nicehash, for example. It's currently at 2114.5 BTC/GH/day - sounds incredible? Well, actually, it isn't. Look at the miners: a RTX3070 will hash at a rate of aprox. 41.6 H/s. Now we do the math, first we need to get 41.6H to GH:  41.6H = 41.6*Pow(10,-9) GH - now we can multiplicate:

Income = 2114.5 BTC/GH/day * 41.6*Pow(10,-9) GH = 0.00008796320 BTC/day

To get a feeling for it, look at "Miners" and "Best Miners" - these two tables contain real income values (pool income x miner hashrate)

> 6) How does the transfer of earned pool work funds to the wallet of choice (XMR for me) work? Are somehow all these coin mining automatically converted? I can see some pool balances (30 USD for example atm) - when and how will these be transferred into XMR?

That depends heavily on the pool(s), you have selected. If you like, show me your current config, and I can check, if the setup is valid.

> 7) Is there a way to make the web interface more dense/compact (config file edit is fine!) and is it possible to see all client servers to a master server in one view without using rbminer?

On the server's API dashboard, all connected miners are listed. Yes, the web interface setup needs some more work. I am on it :)

> 8) To enable all pools and all mining algo's does one simply need to have a BTC wallet and then enable all pools, or are multiple wallets required? Does adding more coins make the overall algorithm more profitable somehow (more options available)?

Depends on the pools, of course. Some pools do autoexchange to BTC, only, others do autoexchange to XMR or multiple wallets. You will need to look at the pools. I have created a list on the readme page.
Hint: do NOT enable all pools. Choose a handful of pools, you trust and wrap your setup around them. Otherwise, you will mine microbitcoins to here and there and get nothing out of it for ages, since most pools have a minimum payout.

> 9) Is the "fast switching" option I saw in the config files of interest to use?

Mostly, no. It's mostly to help in debugging or for some edge cases.

> Thank you

You are welcome!

**Comment by f8vze** [comment]
Thank you @RainbowMiner for a detailed and caring reply. I will review, work on this and may come back with a few followup questions. May also make merge proposals. Thank you very much, and I am getting more and more sure I will choose RainbowMiner as my overall solution. Planning 10 rigs in next 6 months, and will slowly start telling others about it I think. Thank you very much.
